### PR TITLE
[release-4.14] OCPBUGS-29606: Update HostNetworkNamespace address_set for remote zone nodes

### PR DESCRIPTION
**Pull request by arghosh93**
While Interconnect is enabled, it becomes essential to keep address_set specific to HostNetworkNamespace updated when a	remote zone node gets added to the cluster.

This PR is to add/remove management-port IP & node-gateway-router-lrp-ifaddr of both remote and local zone nodes to/from address_set created for HostNetworkNamespace.

Changelog:
- Added a syncMap syncHostNetAddrSetFailed to be used by node event handler. This is required as AddResource event for remote zone nodes are received a lot before than required annotations are set on the node being added.
- addIPToHostNetworkNamespaceAddrsSet is to add management-port IP & node-gateway-router-lrp-ifaddr to address_set while adding both local and remote zone node.
- delIPFromHostNetworkNamespaceAddrsSet is to delete management-port IP & node-gateway-router-lrp-ifaddr from address_set during node deletion for both local and remote zone nodes.
- Above mentioned functions gets called from either AddResource or UpdateResource for NodeType objects.

Jira: https://issues.redhat.com/browse/OCPBUGS-28724
Upstream Issue: https://github.com/ovn-org/ovn-kubernetes/issues/4098

Signed-off-by: Arnab Ghosh <arnabghosh89@gmail.com>
(cherry picked from commit 8cba7dde86c8e1c15320d1327947f27257e0d6ab)

**- What this PR does and why is it needed**

**- Special notes for reviewers**

cherry picked from #2074 

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/pkg/ovn/namespace.go [hunk 1]
```diff
@@ -342,6 +342,18 @@ func (oc *DefaultNetworkController) getAllHostNamespaceAddresses() []net.IP {
...
 	var ips []net.IP
+	// skip hybrid overlay nodes
+	// This is a release-4.14 specific change required as addIPToHostNetworkNamespaceAddrSet()
+	// and delIPFromHostNetworkNamespaceAddrSet() getting executed even for HybridOverlay nodes.
+	// release-4.15 onward below commit is merged to take care of HybridOverlay nodes and return
+	// from either AddResource() or UpdateResource().
+	// https://github.com/openshift/ovn-kubernetes/commit/aeb9817f5634a8ed46873c9afd221114a0344291
+	// Start OCPBUG downstream SPECIFIC
+	if util.NoHostSubnet(node) {
+		klog.Warningf("Node %s is a hybrid overlay node", node.Name)
```
Comment (tssurya) [inline go-controller/pkg/ovn/namespace.go:353]: nit: `klog.Warningf("Node %s is a hybrid overlay node; so not processing it for updating host namespace address-set blah", node.Name)`

**Review by tssurya — COMMENTED** [review]
/lgtm
 no need to fix the nits, leave them, but can you answer my question around conflicts? did you have any compared to 4.15 ?

File: go-controller/pkg/ovn/base_network_controller.go
[the remark this answers is not in the document]
Comment (arghosh93) [reply go-controller/pkg/ovn/base_network_controller.go:1]: @tssurya there were 4 conflicts:
1. https://github.com/arghosh93/ovn-kubernetes-ds/blob/release-4.15/go-controller/pkg/ovn/namespace.go#L331-L334 was not there in release-4.14
2. String inside double quotes was different in release-4.14 than https://github.com/arghosh93/ovn-kubernetes-ds/blob/release-4.15/go-controller/pkg/ovn/master_test.go#L1539 
3. DefaultNetworkController struct(https://github.com/arghosh93/ovn-kubernetes-ds/blob/release-4.15/go-controller/pkg/ovn/default_network_controller.go#L48) did not contain syncMigratablePodsFailed in release-4.15 but release-4.14 had it.
4. release-4.14 did not have https://github.com/arghosh93/ovn-kubernetes-ds/blob/release-4.15/go-controller/pkg/ovn/default_network_controller.go#L742-L751.

**Pull request merged**